src/Interfaces.jl

```diff
@@ -165,10 +165,20 @@ function toimplements!(T, arg::Expr, shouldthrow::Bool=true)
     end
 end
 
-macro interface(T, block)
+macro interface(T, alias_or_block, maybe_block=nothing)
     @assert T isa Symbol || T.head == :.
+    if alias_or_block isa Symbol
+        alias = alias_or_block
+        block = maybe_block
+    else
+        alias = T
+        block = alias_or_block
+    end
     @assert block isa Expr && block.head == :block
     Base.remove_linenums!(block)
+    if T !== alias
+        recursiveswapT!(alias, block, T)
```

Comment:
is this the right way round?

should it be
```diff
- recursiveswapT!(alias, block, T)
+ recursiveswapT!(T, block, alias)
```

Reply:
ohhh, we swap e.g. `X` -> `SomeInterface` here (where `X` is the user-given alias, which may or may not be `T`)
then later down switch `SomeInterface` -> `T`

Reply:
if that's what's going on, might be worth adding a comment 😊

Reply:
and/or maybe our variable `T` could be named `IT` or `InterfaceType` or something,  so make the code easier  to follow? (too many "`T`"s 😆 )

Reply:
Yeah, I was also thinking we should rename `recursivewapT!` to just `recursiveswapsymbols!` or something, since it'd doesn't swap for just `T` anymore